### PR TITLE
Improve mask help text

### LINKED ISSUE #248: Improve mask help text
The help could have more details
i.	Trace, Create mask, Extend Mask (verify in 2D view), Save Mask

Extend: applies the mask to the other slices in the volume. Keep the mask 2D only if you are looking at a section of the object.

### PR DESCRIPTION
**Pull request by DanicaSTFC**
- Edit mask help text 
- Edit documentation
- Added spline curve instructions in both files.

Closes #248 

Help text look:
mask help

### CODE REVIEW

File: src/idvc/dvc_interface.py [hunk 1]
```diff
@@ -428,8 +428,15 @@ def CreateHelpPanel(self):
             "9. Once satisfied with the registration, move to the next tab. Be aware that point 0 is the point from which DVC will start."
             )
         
-        self.help_text.append("To create a mask you need to create a selection. Start tracing a freehand region for the selection by clicking 'Start Tracing' button.\n"
-            "When you are happy with your region click 'Create Mask'.")
+        self.help_text.append(
+            "1. If you already have a mask file, upload it.\n"
+            "2. To create a new mask click 'Start tracing'. Click on the image and trace a freehand region while holding the mouse button.\n"
```
Comment (paskino) [inline src/idvc/dvc_interface.py:433]: How to use the [tracing widget](https://vtk.org/doc/nightly/html/classvtkImageTracerWidget.html#details):

The button actions and key modifiers are as follows for controlling the widget: 

1) left button click over the image, hold and drag draws a free hand line. 
2) left button click and release erases the widget line, if it exists, and repositions the first handle. 
3) middle button click starts a snap drawn line. The line is terminated by clicking the middle button while depressing the ctrl key. 
4) when tracing a continuous or snap drawn line, if the last cursor position is within a specified tolerance to the first handle, the widget line will form a closed loop. 
5) right button clicking and holding on any handle that is part of a snap drawn line allows handle dragging: existing line segments are updated accordingly. The path can be closed by repositioning the first and last points over one another. 
6) ctrl key + right button down on any handle will erase it: existing snap drawn line segments are updated accordingly. If the line was formed by continuous tracing, the line is deleted leaving one handle. 
7) shift key + right button down on any snap drawn line segment will insert a handle at the cursor position. The line segment is split accordingly.

**Review by paskino — COMMENTED** [review]
I don't know if you want to add all of my comment, but that is how the tracer works.

File: src/idvc/dvc_interface.py (hunk 1 above)
Comment (DanicaSTFC) [reply src/idvc/dvc_interface.py:433]: I think we should add these instructions, thanks. I need to think where the best place to insert them would be

Comment (DanicaSTFC) [reply src/idvc/dvc_interface.py]: We discussed this with Brian on 06/06/24 and he also agreed it was a great feature to emphasize.

File: docs/source/dvc_steps.rst [hunk 2]
```diff
@@ -59,31 +59,51 @@ Move to the **Mask** tab only when the registration has been confirmed at least
 
 Mask Creation
 =============
+A mask needs to be created to dictate where the point cloud will lie.
```
Comment (paskino) [inline docs/source/dvc_steps.rst:62]:
```suggestion
The point cloud will be created inside a mask defined by the user. A mask is a binary image where ones represent where the points will lie. iDVC allows you to create or import a mask via file. 
```

File: docs/source/dvc_steps.rst [hunk 3]
```diff
@@ -59,31 +59,51 @@ Move to the **Mask** tab only when the registration has been confirmed at least
...
+Once satisfied with the mask, move on to the **Point Cloud** panel.
 
 Creating a mask
 ~~~~~~~~~~~~~~~
+Click on the **Start Tracing** button to draw a mask and enable freehand tracing on the viewer.
```
Comment (paskino) [inline docs/source/dvc_steps.rst:67]:
```suggestion
A mask is created by tracing the cross section of the mask and extending it above and below the current slice by **Slices Above** and **Slices Below** values. 
More complex masks can be created by extending the mask by multiple tracing.

The user can trace in 2 modalities: free hand and or by inserting multiple segments separated by point handles.

Click on the **Start Tracing** button to draw a mask and enable tracing on the viewer. 
```

File: docs/source/dvc_steps.rst [hunk 4]
```diff
@@ -59,31 +59,51 @@ Move to the **Mask** tab only when the registration has been confirmed at least
...
 Mask Creation
 =============
+A mask needs to be created to dictate where the point cloud will lie.
+Once satisfied with the mask, move on to the **Point Cloud** panel.
 
 Creating a mask
 ~~~~~~~~~~~~~~~
+Click on the **Start Tracing** button to draw a mask and enable freehand tracing on the viewer. 
+The mask design consists of either the tracing of a single curve or the insertion of multiple spline segments separated by point handles.
+The traced line to create a mask can be controlled following the instructions below.
```
Comment (paskino) [inline docs/source/dvc_steps.rst:69]:
```suggestion
```

File: docs/source/dvc_steps.rst [hunk 5]
```diff
@@ -59,31 +59,51 @@ Move to the **Mask** tab only when the registration has been confirmed at least
...
 
-A mask needs to be created to dictate where the point cloud will lie.
-To draw a mask, click on the **Start Tracing** button.
-This will allow you to trace a region freehand by left button clicking and dragging the mouse.
+- Draw a free hand line: left button click over the image, hold and drag.
```
Comment (paskino) [inline docs/source/dvc_steps.rst:71]:
```suggestion
Freehand tracing:
- Draw a free hand line: left button click over the image, hold and drag.
```

File: docs/source/dvc_steps.rst [hunk 6]
```diff
@@ -59,31 +59,51 @@ Move to the **Mask** tab only when the registration has been confirmed at least
...
+- Erase the line: left button click and release.
+- Start a snap drawn line: middle button click. Terminate the line by clicking the middle button while depressing the ctrl key.
```
Comment (paskino) [inline docs/source/dvc_steps.rst:73]:
```suggestion
Multisegment tracing:
- Start a snap drawn line: middle button click. Terminate the line by clicking the middle button while depressing the ctrl key. 
```

File: docs/source/dvc_steps.rst [hunk 7]
```diff
@@ -59,31 +59,51 @@ Move to the **Mask** tab only when the registration has been confirmed at least
...
+The 2D mask drawn in the viewer is used across multiple slices in 3D, above and below the current slice; the volume can be adjusted by editing the **Slices Above** and **Slices Below** values.
+Click on **Create mask** when the tracing is finalised.
+
+Note: the **Slices Above** and **Slices Below** are in the coordinate system of the downsampled image (if the images have been downsampled).
+
+Extending a mask
+~~~~~~~~~~~~~~~~
+
+Tick the **Extend Mask** checkbox if the mask needs to cover more than one area, or the area of the mask needs to be enlarged. 
+Then, draw another region and press the button **Extend Mask**.
```
Comment (paskino) [inline docs/source/dvc_steps.rst:88]:
```suggestion
Then, draw another region and press the button **Extend Mask**.

If **Extend Mask** is not checked the mask will be reset when tracing.
```

**Pull request merged**